crates/walrus-service/src/node/blob_event_processor.rs

```diff
@@ -229,15 +295,22 @@ impl BlobEventProcessor {
     ) -> Self {
         let mut senders = Vec::with_capacity(num_workers);
         let mut workers = Vec::with_capacity(num_workers);
+        let mut background_per_processor_pending_event_count = Vec::with_capacity(num_workers);
         for worker_index in 0..num_workers {
             let (tx, rx) = mpsc::unbounded_channel();
             senders.push(tx);
-            let mut background_processor =
-                BackgroundEventProcessor::new(node.clone(), blob_sync_handler.clone(), rx);
+            let pending_event_count = Arc::new(AtomicU64::new(0));
+            background_per_processor_pending_event_count.push(pending_event_count.clone());
```

Comment:
Nit: I don't think we need the `clone` here.
```suggestion
            background_per_processor_pending_event_count.push(pending_event_count);
```